frontend/public/components/overview.jsx

```diff
@@ -68,33 +70,38 @@ class OverviewDetails extends React.Component {
       filteredItems: [],
       groupedItems: [],
       groupOptions: {},
-      selectedGroupLabel: 'app'
+      selectedGroupLabel: ''
     };
   }
 
   componentDidUpdate(prevProps, prevState) {
-    if (this.props.namespace !== prevProps.namespace
-      || this.props.replicationControllers !== prevProps.replicationControllers
-      || this.props.pods !== prevProps.pods
-      || this.props.deploymentConfigs !== prevProps.deploymentConfigs) {
+    const {deployments, deploymentConfigs, namespace, pods, replicaSets, replicationControllers} = this.props;
+    const {filterValue, selectedGroupLabel} = this.state;
+    if (namespace !== prevProps.namespace
+      || replicationControllers !== prevProps.replicationControllers
+      || replicaSets !== prevProps.replicaSets
+      || pods !== prevProps.pods
+      || deploymentConfigs !== prevProps.deploymentConfigs
+      || deployments !== prevProps.deployments) {
       this.createOverviewData();
-    } else if (this.state.filterValue !== prevState.filterValue) {
+    } else if (filterValue !== prevState.filterValue) {
       const filteredItems = this.filterItems(this.state.items);
       this.setState({
         filteredItems,
-        groupedItems: this.groupByLabel(filteredItems, this.state.selectedGroupLabel)
+        groupedItems: this.groupItems(filteredItems, selectedGroupLabel)
       });
-    } else if (this.state.selectedGroupLabel !== prevState.selectedGroupLabel) {
+    } else if (selectedGroupLabel !== prevState.selectedGroupLabel) {
       this.setState({
-        groupedItems: this.groupByLabel(this.state.filteredItems, this.state.selectedGroupLabel)
+        groupedItems: this.groupItems(this.state.filteredItems, selectedGroupLabel)
       });
     }
   }
 
-  sortByDeploymentVersion(replicationControllers, descending) {
-    const compareDeployments = (left, right) => {
-      const leftVersion = parseInt(_.get(left, 'openshift.io/deployment-config.latest-version'), 10);
-      const rightVersion = parseInt(_.get(right, 'openshift.io/deployment-config.latest-version'), 10);
+  sortReplicatorsByVersion(replicators, kind, descending) {
+    const compare = (left, right) => {
+      const annotation = kind === 'ReplicationController' ? 'openshift.io/deployment-config.latest-version' : 'deployment.kubernetes.io/revision';
```

Comment:
I think it would be a little cleaner not to pass kind here, but to instead pass either the annotation name or a callback for getting the revision. Something like

```js
sortByRevision(replicators, descending, annotation) ...

sortRC(replicationControllers, descending) => sortByRevision(replicationControllers, descending, 'openshift.io/deployment-config.latest-version');

sortRS(replicaSets, descending) => sortByRevision(replicaSets, descending, 'deployment.kubernetes.io/revision');
```